File: dynamic_programming/WordBreak.py
class Solution:
    def wordBreak(self, s: str, wordDict: List[str]) -> bool:
        size = len(s)
        list = [False] * (size + 1)
        list[0] = True
        
        for i in range(size+1):
            if list[i]:
                for j in range (i+1, size+1):
                    if s[i:j] in wordDict:
                        list[j] = True
        return list[-1]

File: dynamic_programming/WordBreak.cs
public class Solution {
    public bool WordBreak(string s, IList<string> wordDict) {
        int size = s.Length;
        var list = Enumerable.Repeat(false,size+1).ToArray();
        list[0] = true;
        
        for(int i=0; i<= s.Length; i++){
            if(list[i])
                for (int j=i+1; j<= s.Length; j++){
                    if(wordDict.Contains(s[i..j]))
                        list[j] = true;
                }
        }
        return list.Last();
    }
}